### PR TITLE
Fix D&D resource display names

### PR DESCRIPTION
**Pull request by jpople**
Closes #HJ-162

### Description Of Changes

Fixes D&D resource display names to be more generalizable across systems and handle some edge cases better by no longer relying (solely) on URN length.

### Steps to Confirm

To test, have some data in your detection or discovery tables with:
* nested fields (I used BigQuery)
* a system without a database layer (I used DynamoDB)

1. View D&D tables
1. At all levels, non-subfield resource names should display the resource's name as-is
1. Subfields should display all names below the top-level field name, separated by `.`

### Pre-Merge Checklist

* [x] Issue requirements met
* [x] All CI pipelines succeeded
* [x] `CHANGELOG.md` updated
* Followup issues:
  * [ ] Followup issues created (include link)
  * [x] No followup issues
* Database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!
  * [x] No migrations
* Documentation:
  * [ ] Documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] Documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] If there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
  * [x] No documentation updates required

### CODE REVIEW

File: clients/admin-ui/src/features/data-discovery-and-detection/utils/getResourceName.ts [hunk 1]
```diff
@@ -1,22 +1,47 @@
...
+  const splitUrn = urn.split(URN_SEPARATOR);
 
-const getResourceName = (resource: DiscoveryMonitorItem) => {
-  const URN_SEPARATOR = ".";
-  const splitUrn = resource.urn.split(URN_SEPARATOR);
   if (
-    !resource.parent_table_urn ||
-    splitUrn.length === TOP_LEVEL_FIELD_URN_PARTS
+    !table_name ||
+    !top_level_field_name ||
```
Comment (thingscouldbeworse) [inline clients/admin-ui/src/features/data-discovery-and-detection/utils/getResourceName.ts:21]: If it would be helpful I am adding in `top_level_field_urn` as well in this PR: https://github.com/ethyca/fidesplus/pull/1727/files#diff-8bd50e59212bea55bbd599dd3564999eecbda0c718c2681318bbc990a29d28c9R354

not sure if it's actually better or the same as `top_level_field_name` but I figured better to give as much info as possible now and then we can remove one later if it's not helpful

Comment (thingscouldbeworse) [reply clients/admin-ui/src/features/data-discovery-and-detection/utils/getResourceName.ts:21]: I imagine `top_level_field_name` to be maybe a little tricky in structures like 
```
same_name
                  |------- same_name
                                                 |------- same_name
                                                                                 |------- same_name
```
etc etc

Comment (jpople) [reply clients/admin-ui/src/features/data-discovery-and-detection/utils/getResourceName.ts:21]: Thanks for flagging-- yeah, that'll make this much cleaner if I can just elide that.

Comment (thingscouldbeworse) [reply clients/admin-ui/src/features/data-discovery-and-detection/utils/getResourceName.ts:21]: `top_level_field_urn` is in now btw
